wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+            e_obj.add_field(name="Notification DM", value="SENT\n" if dm else "NOT SENT, DUE TO USER DM PREF's\n",
+                            inline=False)
+            e_obj.set_footer(text="Moderator Action")
+            if e_obj:
+                e_obj.timestamp = dt.astimezone(pytz.timezone('Canada/Pacific'))
+                await self.mod_channel.send(embed=e_obj)
+            logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {username}.")
+
+            # update db
+            await BannedUsers.insert_ban(username, user.id)
+            await BanRecords.insert_record(username, user.id, mod_info[0], mod_info[1], dt, reason)
+
+    @commands.command()
+    async def unban(self, ctx, _id: int):
+        """Unbans a user"""
+
+        logger.info(f"[Ban unban()] unban command detected from {ctx.author} with args=[ {_id} ]")
+        if _id not in self.ban_list:
+            logger.info(f"Provided id: {_id}, does not belong to a banned member.")
+            e_obj = await em(ctx, title="Error",
+                             content=[("Problem",
+                                       f"`{_id}` is either not a valid Discord ID **OR** is not a banned user.")],
+                             colour=self.errorColour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        # "unban"
+        self.ban_list.remove(_id)
+
+        name = await BannedUsers.del_banned_user_by_id(_id)
+
+        logger.info(f"[Ban unban()] User: {name} with id: {_id} was unbanned.")
+        e_obj = await em(ctx, title="Unban", description=f"**`{name}`** was unbanned.", colour=discord.Color.red())
+        if e_obj:
+            await self.mod_channel.send(embed=e_obj)
+
+    @unban.error
+    async def unban_error(self, ctx, error):
+        """Catches an error in unban when a non integer is passed in as an argument"""
+
+        logger.info("[Ban Unban_error] caught non integer ID passed into unban parameter. Handled accordingly")
+        if isinstance(error, commands.BadArgument):
+            e_obj = await em(ctx, title="Error",
+                             content=[("Problem", "Please enter a numerical Discord ID.")],
+                             colour=self.errorColour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+
+    @commands.command()
+    async def bans(self, ctx):
+        """Gets all banned users"""
+
+        logger.info(f"[Ban bans()] bans command detected from {ctx.author}")
+
+        bans = await BannedUsers.get_all_bans()
+        logger.info(f"[Ban bans()] retrieved all banned users: {bans}")
+
+        emb = discord.Embed(title="Banned members", color=discord.Color.red())
+
+        names = ""
+        ids = ""
+        for row in bans:
+            name = row[0]
```

Comment:
you should not need to refer to results in an entry from the DB via row indexing....